### PR TITLE
Create contact and subscription

### PR DESCRIPTION
**Pull request by gsvr**
Create seperate contacts for the CHEW and Mama as soon as their numbers become available for tracking metrics.
When the registration finishes, update the Mama's contact and create a subscription.

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: src/voice_registration.js [hunk 1]
```diff
@@ -44,7 +56,13 @@ go.app = function() {
                     if (go.utils.check_valid_phone_number(content) === false) {
                         return 'state_r02_retry_number';
                     } else {
-                        return 'state_r03_receiver';
+                        return go.utils
+                            // get or create mama contact
+                            .get_or_create_contact('+'+content, self.im)
```
Comment (imsickofmaps) [inline src/voice_registration.js]: I'd prefer we remove the `+` here and move any address normalisation to the utils

File: test/fixtures.js [hunk 2]
```diff
@@ -1,37 +1,377 @@
...
     return [
-        // create contact 082001 Johnny
+        // get contact 07030010001 by msisdn
         {
+
             'request': {
-                'method': 'POST',
+                'method': 'GET',
+                'params': {
+                    'to_addr': '+07030010001'
```
Comment (imsickofmaps) [inline test/fixtures.js]: param should be `msisdn`

File: test/fixtures.js [hunk 3]
```diff
@@ -1,37 +1,377 @@
...
                     'Authorization': ['Token test_key'],
                     'Content-Type': ['application/json']
                 },
-                'url': 'http://localhost:8000/api/v1/contacts/',
-                'data': {
+                'url': 'http://localhost:8000/api/v1/contacts/search/',
+            },
+            'response': {
+                "code": 200,
+                "data": [
```
Comment (imsickofmaps) [inline test/fixtures.js]: response data:`{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "url": "http://localhost:8000/api/v1/contacts/376c67e6-273d-491e-a1fa-e5915415c47a/",
            "id": "376c67e6-273d-491e-a1fa-e5915415c47a",
            "version": 1,
            "details": {
                "default_addr_type": "msisdn",
                "addresses": "msisdn:+07030010001"
            },
            "created_at": "2015-06-12T16:01:43.601992Z",
            "updated_at": "2015-07-22T15:50:43.449693Z"
        }
    ]
}`

File: src/utils.js [hunk 4]
```diff
@@ -146,22 +147,266 @@ go.utils = {
...
+    get_contact_id_by_msisdn: function(msisdn, im) {
+        var params = {
+            to_addr: msisdn
+        };
+        return go.utils
+            .control_api_call('get', params, null, 'contacts/search/', im)
+            .then(function(json_get_response) {
+                var contacts_found = json_get_response.data;
+                // Return the first contact's id
+                return contacts_found[0].id || "no_contacts_found";
```
Comment (imsickofmaps) [inline src/utils.js]: this function should return `null` or `false` if it doesn't find anything

Comment (gsvr) [inline src/utils.js]: so you don't like my "no_contacts_found" string?

Comment (imsickofmaps) [inline src/utils.js]: no

**Comment by imsickofmaps** [comment]
:+1:

**Pull request closed without merging**